src/update.lua

```diff
@@ -0,0 +1,52 @@
+--[[
+    This part of the plugin is used to fix possible errors when users update the Plugin or install it for the first time.
+]]
+
+local Plugin : Plugin
+local stringify = require(script.Parent.stringify)
+
+local currentVersion = "0.0.0" -- the current version of the plugin.
+
+local updateFunctions = {
+    function()
+        -- Fresh install / before 0.0.0
+        local uid = Plugin:GetSetting("user-id")
+        local uid2 = Plugin:GetSetting("UserId")
+        local tok = Plugin:GetSetting("refresh-token")
+        local ses = Plugin:GetSetting("session")
+        
+        local baseTable = {}
+        if uid or uid2 then
+            baseTable["user-id"] = uid or uid2
+            Plugin:SetSetting("user-id", nil)
+            Plugin:SetSetting("UserId", nil)
+        end
+        if tok then
+            baseTable["refresh-token"] = tok
+            Plugin:SetSetting("refresh-token", nil)
+        end
+        if ses then
+            baseTable["session"] = ses
+            Plugin:SetSetting("session", nil)
+        end
+        if baseTable ~= {} then
+            Plugin:SetSetting("auth", "!" .. stringify(baseTable))
+        else
+            Plugin:SetSetting("auth", "!{}")
+        end
+
+    end
+}
+
+local function update(pl)
+    Plugin = pl
+    if Plugin:GetSetting("pluginVersion") ~= "#" .. currentVersion then
+        for i, update in pairs(updateFunctions) do
```

Comment:
We should only run functions which are specified for the current version or newer. This is not only to prevent re-running, but to prevent old updates from ruining newer data. Versions with updates will need to be in an ordered list, and then there should be an update function associated with each version.

Reply:
Alright. adding that to the new commit.